File: backend/src/main/java/ru/morzevichka/competition_system/model/Pilot.java
package ru.morzevichka.competition_system.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Entity
@Builder
@Table(name = "pilots")
@AllArgsConstructor
@NoArgsConstructor
public class Pilot {

    @Id
    @GeneratedValue(strategy= GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, unique = true)
    private String nickname;

    @Column(name = "first_name", nullable = false)
    private String firstName;

    @Column(name = "last_name", nullable = false)
    private String lastName;

    @Column(name = "tg_username", unique = true)
    private String tgUsername;
}

File: frontend/app/types/pilot.ts
export interface Pilot {
    id: string;
    nickname: string;
    first_name: string;
    last_name: string;
    tg_username: string;
}

export interface CreatePilotRequest {
    nickname: string;
    first_name: string;
    last_name: string;
    tg_username: string;
}

export interface UpdatePilotRequest {
    id: string;
    nickname: string;
    first_name: string;
    last_name: string;
    tg_username: string;
}